remote/github/github.go

```diff
@@ -330,3 +346,162 @@ func (g *Github) GetHook(r *http.Request) (*model.Hook, error) {
 
 	return hook, nil
 }
+
+func (g *Github) GetStatusHook(r *http.Request) (*model.StatusHook, error) {
+
+	// only process comment hooks
+	if r.Header.Get("X-Github-Event") != "status" {
+		return nil, nil
+	}
+
+	data := statusHook{}
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debug(data)
+
+	if data.State != "success" {
+		return nil, nil
+	}
+
+	hook := new(model.StatusHook)
+
+	hook.SHA = data.SHA
+
+	hook.Repo = new(model.Repo)
+	hook.Repo.Owner = data.Repository.Owner.Login
+	hook.Repo.Name = data.Repository.Name
+	hook.Repo.Slug = data.Repository.FullName
+
+	log.Debug(*hook)
+
+	return hook, nil
+}
+
+func (g *Github) GetPullRequestsForCommit(u *model.User, r *model.Repo, sha *string) ([]model.PullRequest, error) {
+	client := setupClient(g.API, u.Token)
+	fmt.Println("sha == ", sha, *sha)
+	issues, _, err := client.Search.Issues(fmt.Sprintf("%s&type=pr", *sha), &github.SearchOptions {
+		TextMatch: false,
+	})
+	if err != nil {
+		return nil, err
+	}
+	out := make([]model.PullRequest, len(issues.Issues))
+	for k, v := range issues.Issues {
+		pr, _, err := client.PullRequests.Get(r.Owner, r.Name, *v.Number)
+		if err != nil {
+			return nil, err
+		}
+
+		mergeable:= true
+		if pr.Mergeable != nil {
+			mergeable = *pr.Mergeable
+		}
+
+		status, _, err := client.Repositories.GetCombinedStatus(r.Owner, r.Name, *sha, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		out[k] = model.PullRequest{
+			Issue: model.Issue{
+				Number: *v.Number,
+				Title: *v.Title,
+				Author: *v.User.Login,
+			},
+			Branch:  model.Branch {
+				Name: *pr.Head.Ref,
+				BranchStatus: *status.State,
+				Mergeable: mergeable,
+
+			},
+		}
+	}
+	return out, nil
+}
+
+func (g *Github) GetBranchStatus(u *model.User, r *model.Repo, branch string) (*model.BranchStatus, error) {
+	client := setupClient(g.API, u.Token)
+	statuses, _, err := client.Repositories.GetCombinedStatus(r.Owner, r.Name, branch, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return (*model.BranchStatus)(statuses.State), nil
+}
+
+func (g *Github) MergePR(u *model.User, r *model.Repo, pullRequest model.PullRequest) (*string, error) {
+	client := setupClient(g.API, u.Token)
+
+	result, _, err := client.PullRequests.Merge(r.Owner, r.Name, pullRequest.Number, "Merged by LGTM")
+	if err != nil {
+		return nil, err
+	}
+
+	if !(*result.Merged) {
+		return nil, errors.New(*result.Message)
+	}
+	return result.SHA, nil
+}
+
+func (g *Github) GetMaxExistingTag(u *model.User, r *model.Repo) (*version.Version, error) {
```

Comment:
for the helper function, perhaps follow a pattern similar to this, adding a `GetMaxTag` helper to the remote package that wraps `GetTagList`
https://github.com/lgtmco/lgtm/blob/master/cache/helper.go